models.py

```diff
@@ -81,6 +81,20 @@ def seq2seq(encoder_vocab_size, encoder_maxlen,
 
     return model
 
+# function for Lambda layers
+def last_layer(x):
+    get_last_layer = lambda x:x[:,-1,:]
+    return get_last_layer(x)
+
+def last_layer_output(input_shape):
+    return (input_shape[0], input_shape[2])
+
+def sum_tensor(x):
+    return K.sum(x, axis=1)
+
+def sum_tensor_output(input_shape):
+    return (input_shape[0],input_shape[2])
+
```

Comment:
The function "sum_tensor_output" is the same as "last_layer_output".
Duplicated code should be removed.